react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+
+    };
+
+    this.slideChange = this.slideChange.bind(this);
+  }
+
+  slideChange(event) {
+    const { urls } = this.props;
+    const { url } = this.state;
+
+    let nextUrl;
+
+    if (url !== undefined) {
+      urls.forEach((item, index) => {
+        if (item === url) {
+          if (event === undefined || event.target.getAttribute('action') === 'next') {
+            nextUrl = urls[index + 1];
+          } else if (event.target.getAttribute('action') === 'previous') {
+            if (index !== 0) {
+              nextUrl = urls[index - 1];
+            } else {
+              nextUrl = urls[urls.length - 1];
+            }
+          } else if (event.target.getAttribute('action') === 'target') {
+            const selectionIndex = event.target.getAttribute('selection');
+            nextUrl = urls[selectionIndex];
+          }
+        }
+      });
+    } else {
+      nextUrl = urls[1];
+    }
+
+    this.setState({ url: nextUrl });
+    if (event !== undefined) {
+      const timeoutId = event.target.getAttribute('timeoutid');
+      clearTimeout(timeoutId);
+    }
+
+  }
```

Comment:
While it looks like this probably works it would be a bit simpler to make separate methods for going to the next image, the previous image, and going to a specific index. You can then use the specific method on the specific element that needs it. This would simplify the logic a bit.